### PR TITLE
Add new APICompat TOC node and tool article

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to #36235 

- Moves Package validation under APICompat
- Adds new article about Microsoft.DotNet.ApiCompat.Tool global tool
- Adds more suppression info

Previews:

- [Microsoft.DotNet.ApiCompat.Tool global tool](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/global-tool?branch=pr-en-us-38470) 
- [Package validation error codes](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/diagnostic-ids?branch=pr-en-us-38470)

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/core-libraries/5.0/code-access-security-apis-obsolete.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/core/compatibility/core-libraries/5.0/code-access-security-apis-obsolete.md) | [Most code access security APIs are obsolete](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/5.0/code-access-security-apis-obsolete?branch=pr-en-us-38470) |
| [docs/core/project-sdk/msbuild-props.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/core/project-sdk/msbuild-props.md) | [MSBuild properties for Microsoft.NET.Sdk](https://review.learn.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props?branch=pr-en-us-38470) |
| [docs/core/whats-new/dotnet-6.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/core/whats-new/dotnet-6.md) | [What's new in .NET 6](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-6?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/global-tool.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/global-tool.md) | [Microsoft.DotNet.ApiCompat.Tool global tool](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/global-tool?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/package-validation/baseline-version-validator.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/package-validation/baseline-version-validator.md) | [docs/fundamentals/apicompat/package-validation/baseline-version-validator](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/baseline-version-validator?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/package-validation/compatible-framework-in-package-validator.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/package-validation/compatible-framework-in-package-validator.md) | [Validate compatible frameworks](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/compatible-framework-in-package-validator?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/package-validation/compatible-framework-validator.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/package-validation/compatible-framework-validator.md) | [Validating Different Runtimes](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/compatible-framework-validator?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/package-validation/diagnostic-ids.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/package-validation/diagnostic-ids.md) | [Package validation error codes](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/diagnostic-ids?branch=pr-en-us-38470) |
| [docs/fundamentals/apicompat/package-validation/overview.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/fundamentals/apicompat/package-validation/overview.md) | [docs/fundamentals/apicompat/package-validation/overview](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/overview?branch=pr-en-us-38470) |
| [docs/whats-new/dotnet-7-docs.md](https://github.com/dotnet/docs/blob/b36b6263bb837258e3e001ac06f6be14d506c7ce/docs/whats-new/dotnet-7-docs.md) | [.NET docs: What's new for the .NET 7 release](https://review.learn.microsoft.com/en-us/dotnet/whats-new/dotnet-7-docs?branch=pr-en-us-38470) |

### CODE REVIEW

File: docs/fundamentals/apicompat/global-tool.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+title: Microsoft.DotNet.ApiCompat.Tool global tool
+description: Learn about the Microsoft.DotNet.ApiCompat.Tool global tool, which performs API compatibility checks on assemblies and packages.
+ms.date: 11/27/2023
+---
+
+# Microsoft.DotNet.ApiCompat.Tool global tool
+
+The Microsoft.DotNet.ApiCompat.Tool tool performs API compatibility checks on assemblies and packages.
```
Comment (gewarren) [inline docs/fundamentals/apicompat/global-tool.md:9]: I'd like to add more detail here if you have any suggestions.

File: docs/fundamentals/apicompat/global-tool.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+  Specifies a transformation pattern for the *left side* assemblies.
+
+- **`--right-assemblies-transformation-pattern <PATTERN>`**
+
+  Specifies a transformation pattern for the *right side* assemblies.
+
+## Example
+
+The following command compares the .NET 7 and .NET 8 versions of an assembly.
```
Comment (gewarren) [inline docs/fundamentals/apicompat/global-tool.md:127]: Is this a realistic example? What other examples should I add?

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

I'll defer to others on your questions. I don't have better answers.

File: docs/fundamentals/apicompat/global-tool.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+  Enables the rule that checks whether parameter names have changed in public methods.
+
+- **`-l, --left, --left-assembly <PATH>`**
+
+  (REQUIRED) Specifies the path to one or more assemblies that serve as the *left side* to compare.
+
+- **`-r, --right, --right-assembly`**
+
+  (REQUIRED) Specifies the path to one or more assemblies that serve as the *right side* to compare.
```
Comment (ViktorHofer) [inline docs/fundamentals/apicompat/global-tool.md:99]: The CLI tool (and the msbuild task) have two modes:
1. Compare a package against a baseline package or
2. Compare an assembly against a baseline assembly

Some of these options only apply to one mode and others only apply to the other mode. I.e. the Required mention here only applies when comparing assemblies and not packages.

These are the options that only apply to package comparison: https://github.com/dotnet/sdk/blob/b6d7f63756a46fbba9adc6fa360a702ae0e4df4b/src/Compatibility/ApiCompat/Microsoft.DotNet.ApiCompat.Tool/Program.cs#L281-L290

**Review by ViktorHofer — COMMENTED** [review]
This is great. Thank you so much for working on this.

**Pull request merged**